cipher/rsa.c

```diff
@@ -183,14 +183,76 @@ test_keys_fips (gcry_sexp_t skey)
 {
   int result = -1; /* Default to failure.  */
   char plaintext[128];
+  //char plaintext[] =
+  //  "Jim quickly realized that the beautiful gowns are expensive.";
   gcry_sexp_t sig = NULL;
   const char *data_tmpl = "(data (flags pkcs1) (hash %s %b))";
   gcry_md_hd_t hd = NULL;
   int ec;
+  gcry_error_t err;
+  gcry_sexp_t plain = NULL;
+  gcry_sexp_t encr  = NULL;
+  gcry_mpi_t  ciphertext = NULL;
+  gcry_sexp_t decr  = NULL;
+  char *decr_plaintext = NULL;
+  int decr_plaintext_len = 0, strip;
+  gcry_sexp_t tmplist = NULL;  
+  gcry_mpi_t ref_mpi = NULL;
+  int counter;
 
-  /* Create a random plaintext.  */
-  _gcry_randomize (plaintext, sizeof plaintext, GCRY_WEAK_RANDOM);
+    /* Create a random plaintext.  */
```

Comment:
@jallisonciq This update to test_keys_fips doesn't look right, but I think it was added to support the rsa PTC enc/dec feature

Reply:
looks like the code was taken from https://github.com/ciq-rocky-fips/libgcrypt/blob/9b90cfd7f1a94cf79103f5ce787297856c617042/cipher/rsa.c#L2196